### PR TITLE
Feature/security

### PR DESCRIPTION
**Pull request by gaydukov**

### CODE REVIEW

File: src/main/java/com/softensy/SoftensySpringBoot/configuration/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                .antMatchers("/").permitAll()
+                .anyRequest()
+                .authenticated()
+                .and()
+                .httpBasic();
+    }
+
+    @Bean
+    protected PasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder(12);
```
Comment (shelsoftensy) [inline src/main/java/com/softensy/SoftensySpringBoot/configuration/SecurityConfig.java:36]: why here is 12, what default value is?

File: src/main/java/com/softensy/SoftensySpringBoot/configuration/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .csrf().disable()
+                .authorizeRequests()
+                .antMatchers("/").permitAll()
+                .anyRequest()
+                .authenticated()
+                .and()
+                .httpBasic();
```
Comment (shelsoftensy) [inline src/main/java/com/softensy/SoftensySpringBoot/configuration/SecurityConfig.java:31]: httpBasic, why do you use it here?

File: src/main/java/com/softensy/SoftensySpringBoot/dto/UserSecurityDetails.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return true;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return true;
+    }
```
Comment (shelsoftensy) [inline src/main/java/com/softensy/SoftensySpringBoot/dto/UserSecurityDetails.java]: do you need all this boilerplate code?

File: src/main/java/com/softensy/SoftensySpringBoot/entity/DoctorSecurity.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToOne;
+
+@Entity
+@Data
+@SuperBuilder
+@AllArgsConstructor
+@NoArgsConstructor
+public class DoctorSecurity extends UserSecurity {
```
Comment (shelsoftensy) [inline src/main/java/com/softensy/SoftensySpringBoot/entity/DoctorSecurity.java]: if your app will have more entity types, you will add  more entitySecurity classes?

File: src/main/java/com/softensy/SoftensySpringBoot/mapper/UserSecurityMapper.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserSecurityMapper {
+    public UserDetails userSecurityToUserDetails(UserSecurity userSecurity) {
+        return new User(userSecurity.getLogin(), userSecurity.getPassword(),
+                true, true, true, true,
+                userSecurity.getRole().getAuthorities());
```
Comment (shelsoftensy) [inline src/main/java/com/softensy/SoftensySpringBoot/mapper/UserSecurityMapper.java]: soup in constructor values? what is it?

**Review by shelsoftensy — CHANGES_REQUESTED** [review]

File: src/main/java/com/softensy/SoftensySpringBoot/configuration/SecurityConfig.java (hunk 1 above)
Comment (gaydukov) [reply src/main/java/com/softensy/SoftensySpringBoot/configuration/SecurityConfig.java:36]: default value is 10

File: src/main/java/com/softensy/SoftensySpringBoot/dto/UserSecurityDetails.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+@AllArgsConstructor
+@Data
+public class UserSecurityDetails implements UserDetails {
+    private final String username;
+    private final String password;
+    private final List<SimpleGrantedAuthority> authorities;
+    private final boolean isAccountNonExpired;
+    private final boolean isAccountNonLocked;
+    private final boolean isCredentialsNonExpired;
+    private final boolean isEnabled;
```
Comment (shelsoftensy) [inline src/main/java/com/softensy/SoftensySpringBoot/dto/UserSecurityDetails.java]: yagni?

File: src/main/java/com/softensy/SoftensySpringBoot/service/serviceImpl/UserDetailsServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class UserDetailsServiceImpl implements UserDetailsService {
+    private final AppointmentRepository appointmentRepository;
```
Comment (shelsoftensy) [inline src/main/java/com/softensy/SoftensySpringBoot/service/serviceImpl/UserDetailsServiceImpl.java]: appointmentRepo in userDetails service???

File: src/main/java/com/softensy/SoftensySpringBoot/service/serviceImpl/UserDetailsServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+        }
+    }
+
+    public boolean hasAuthorityDoctorInAppointment(long id) {
+        String login = SecurityContextHolder.getContext().getAuthentication().getName();
+        UserSecurity user = userSecurityRepository.findByLogin(login).orElseThrow(() ->
+                new UsernameNotFoundException("User doesn't exists"));
+        Appointment appointment = appointmentRepository.findById(id).orElseThrow(() ->
+                new UsernameNotFoundException("Appointment doesn't exists"));
+        if (user.getAccountType() != AccountType.DOCTOR) {
```
Comment (shelsoftensy) [inline src/main/java/com/softensy/SoftensySpringBoot/service/serviceImpl/UserDetailsServiceImpl.java]: constant value should be at the first place

File: src/test/java/com/softensy/SoftensySpringBoot/TestDataGenerator.java [hunk 9]
```diff
@@ -252,4 +255,38 @@ public static PatientAppointmentDto getThirdPatientAppointmentDto() {
...
+        Set<SimpleGrantedAuthority> authority = new HashSet<>();
+        authority.add(new SimpleGrantedAuthority("appointment:read"));
+        authority.add(new SimpleGrantedAuthority("appointment:write"));
+        authority.add(new SimpleGrantedAuthority("patient:read"));
+        authority.add(new SimpleGrantedAuthority("patient:write"));
+        return authority;
+    }
+
+    public static UserDetails getUserDetails() {
+        return new User("user", "user", true, true, true, true,
```
Comment (shelsoftensy) [inline src/test/java/com/softensy/SoftensySpringBoot/TestDataGenerator.java]: emm

File: src/test/java/com/softensy/SoftensySpringBoot/controller/AppointmentControllerTest.java [hunk 10]
```diff
@@ -83,11 +108,41 @@ void testGetAppointmentToDoctorByDoctorIdReturnStatus200andListPatientsAppointme
     }
 
     @Test
-    @DisplayName("checking get list patient appointments to doctors by patient id with status 200")
-    void testGetPatientAppointmentToDoctorsByPatientIdReturnStatus200andListDoctorsAppointment() throws Exception {
+    @DisplayName("checking forbidden get list appointments to doctor by doctor id with another authenticated with status 403")
+    @WithMockUser(authorities = "patient:read")
+    void testGetAppointmentToDoctorByDoctorIdWithAnotherAuthenticatedReturnStatus403() throws Exception {
```
Comment (shelsoftensy) [inline src/test/java/com/softensy/SoftensySpringBoot/controller/AppointmentControllerTest.java]: AppointmentToDoctor?
english?

**Review by shelsoftensy — CHANGES_REQUESTED** [review]

**Pull request merged**